src/sentry/middleware/user.py

```diff
@@ -1,17 +1,19 @@
 from __future__ import absolute_import
 
-from django.utils import timezone
 from datetime import timedelta
-import inspect
 
+from django.utils import timezone
+from sentry.api.authentication import TokenAuthentication
 
-class UserActiveMiddleware(object):
-    allowed_paths = (
-        'sentry.api.endpoints',
-        'sentry.web.frontend',
-    )
 
+class UserActiveMiddleware(object):
     def process_view(self, request, view_func, view_args, view_kwargs):
+        try:
+            auth = TokenAuthentication()
+            request.user = auth.authenticate(request)[0]
```

Comment:
Ok so:

- if request.user is an authenticated user when they hit an API request thats fired via the javascript app (e.g. loading the stream), then we dont need to do anything special here
- if request.user is filled in later by SessionAuthentication in the API, we should handle it there
- in either situation, we dont care about TokenAuth (I forgot what it meant on original review), and we owuldn't want to approach capturing the api auth here even if we did